### PR TITLE
File and Image field type support

### PR DESCRIPTION
**Pull request by Sergey-Orlov**
added support from Image and File field types

More info at https://projects.skilld.cloud/skilld/ski-contrib/issues/114#note_323381

### CODE REVIEW

File: src/Diagram.php [hunk 1]
```diff
@@ -132,6 +132,9 @@ public function create($entity_types) {
                 $field_property_info['type'] = 'list<' . $field_property_info['type'] . '>';
               }
 
+              $reference_target_type = NULL;
+              $target_bundles = [];
+              // Get target info from entity reference field type.
               if ($instance_info->getType() == 'entity_reference' || $instance_info->getType() == 'entity_reference_revisions') {
```
Comment (piggito) [inline src/Diagram.php]: Here is if for `$instance_info->getType()`

File: src/Diagram.php [hunk 2]
```diff
@@ -144,20 +147,28 @@ public function create($entity_types) {
...
-                      'bundle' => $target_bundle,
-                      'cardinality' => $instance_info->getFieldStorageDefinition()->getCardinality(),
-                      'required' => $instance_info->isRequired(),
-                      'fieldname' => $field_name,
-                    ];
-                  }
+              }
+              else {
+                // Get target info for other field types.
+                switch ($instance_info->getType()) {
```
Comment (piggito) [inline src/Diagram.php]: And in else clause we have a switch also on `$instance_info->getType()`

**Review by piggito — CHANGES_REQUESTED** [review]
This looks a bit disorganized, I think it will be better to have an outer `switch ($instance_info->getType())` and have cases for entity_reference, entity_reference_revisions, image and file so please refactor this code.

File: src/Diagram.php (hunk 1 above)
Comment (Sergey-Orlov) [reply src/Diagram.php]: updated

File: src/Diagram.php (hunk 2 above)
Comment (Sergey-Orlov) [reply src/Diagram.php]: updated

**Review by piggito — COMMENTED** [review]
LGTM just needs QA review

**Comment by skondratieva** [comment]
On my side result is NOT OK. To test I 
- cloned https://github.com/skilld-labs/entity_relationship

**Comment by skondratieva** [comment]
On my side result is NOT OK. To test I 
- cloned https://github.com/skilld-labs/entity_relationship 
- switched to `file_field_support` branch
- added `entity_relationship` module

**Comment by skondratieva** [comment]
On my side result is NOT OK. To test I 
- cloned https://github.com/skilld-labs/entity_relationship 
- switched to `file_field_support` branch
- added `entity_relationship` module to SDC
- built SDC 
- enabled `Entity relationships diagram` module
- on `/admin/reports/diagram` page selected `Content` and submitted form

**Comment by skondratieva** [comment]
On my side result is NOT OK. To test I 
- cloned https://github.com/skilld-labs/entity_relationship 
- switched to `file_field_support` branch
- added `entity_relationship` module to SDC
- built SDC 
- enabled `Entity relationships diagram` module
- on `/admin/reports/diagram` page selected `Content` and submitted form 

selected_8485

**Pull request merged**